lib/cn/spill.h

```diff
@@ -40,7 +42,19 @@ struct cn_compaction_work;
  */
 /* MTF_MOCK */
 merr_t
-cn_spill(struct cn_compaction_work *w);
+cn_spill(
+    struct cn_compaction_work *w,
+    void                      *ctx,
+    uint64_t                   node_dgen,
+    void                      *ekey,
+    uint                      eklen,
+    bool                      *added);
+
+merr_t
+cn_spill_init(struct cn_compaction_work *w, void **ctx);
```

Comment:
Give the context a type name instead of `void *`.  All you need to do is declare `struct spill_ctx;` In this file and then use `struct cpill_ctx *` in place of `void *` where appropriate.